src/ros_main.cpp

```diff
@@ -148,7 +151,8 @@ int main(int argc, char **argv)
   std::transform(args.joint_names.begin(), args.joint_names.end(), args.joint_names.begin(),
                  [&args](std::string name) { return args.prefix + name; });
 
-  std::string local_ip(getLocalIPAccessibleFromHost(args.host));
+  std::string local_ip(args.reverse_ip_address.empty() ? getLocalIPAccessibleFromHost(args.host) :
+                                                         args.reverse_ip_address);
```

Comment:
Suggestion:

```c++
std::string local_ip(args.reverse_ip_address);
if (local_ip.empty())
{
  local_ip = getLocalIPAccessibleFromHost(args.host);
}
```

Ternary expressions aren't very readable.

A regular `if` is also more in-line with the style of the rest of the code.

Reply:
Would be nice to be defensive and check whether `local_ip` is still empty after the `if`. If it is, raise an error.

Reply:
Thanks for the review @gavanderhoorn ! 

Following your suggestion, I changed these bits to:
```c
  std::string local_ip(args.reverse_ip_address);

  if (local_ip.empty())
  {
    local_ip = getLocalIPAccessibleFromHost(args.host);
  }

  if (local_ip.empty())
  {
    LOG_ERROR("reverse_ip_address parameter is not set and automatic detection returned empty!");
    return EXIT_FAILURE;
  }
```
but I noticed that this affects the existing "retry behavior", i.e if the robot is *not* online on launch, the `local_ip` detection fails and now it just exit the whole process. This changes the current behavior that actually keeps retrying with an increasing delay between retries. What would you recommend to do: a) remove the `EXIT_FAILURE` and leave the message as a LOG_WARN only? b) keep the `EXIT_FAILURE`, c) something else? :)

Thx!